xdsl/dialects/mpi.py

```diff
@@ -90,26 +82,33 @@ class ISend(MPIBaseOp):
 
     ## Our Abstraction:
 
-        - We Summarize buf, count and datatype by using memrefs
-        - We assume that tag is compile-time constant
-        - We omit the possibility of using multiple communicators
+        - We omit the possibility of using multiple communicators, defaulting
+          to MPI_COMM_WORLD
     """
 
     name = 'mpi.isend'
 
-    buffer: Annotated[Operand, MemRefType[AnyNumericType]]
+    buffer: Annotated[Operand, Attribute]
```

Comment:
Don't you just want LLVMPointerType here?

Reply:
There are several more of these.

Reply:
We want `llvm.prt` or `fir.ptr`, which are functionally identical, but sadly completely incompatible as far as clan/flang are concerned. Since we don't have the FIR dialect in this repo, our only recourse (for now) is `Attribute`.

But Nick and I will work on a proper abstraction once the paper deadline is over. (and maybe even to get `fir` into xDSL if I dare say so?)

Reply:
Okay, perfect :+1: